src/Mvc/Mvc.Analyzers/src/Startup/StartupAnalzyer.cs

```diff
@@ -12,44 +11,16 @@ namespace Microsoft.AspNetCore.Analyzers
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     public partial class StartupAnalzyer : DiagnosticAnalyzer
     {
-#pragma warning disable RS1008 // Avoid storing per-compilation data into the fields of a diagnostic analyzer.
-        private readonly static Func<StartupAnalysisContext, StartupComputedAnalysis>[] ConfigureServicesMethodAnalysisFactories = new Func<StartupAnalysisContext, StartupComputedAnalysis>[]
-        {
-            ServicesAnalysis.CreateAndInitialize,
-            MvcOptionsAnalysis.CreateAndInitialize,
-        };
-
-        private readonly static Func<StartupAnalysisContext, StartupComputedAnalysis>[] ConfigureMethodAnalysisFactories = new Func<StartupAnalysisContext, StartupComputedAnalysis>[]
-        {
-            MiddlewareAnalysis.CreateAndInitialize,
-        };
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => Diagnostics.SupportedDiagnostics;
 
-        private readonly static Func<CompilationAnalysisContext, ConcurrentBag<StartupComputedAnalysis>, StartupDiagnosticValidator>[] DiagnosticValidatorFactories = new Func<CompilationAnalysisContext, ConcurrentBag<StartupComputedAnalysis>, StartupDiagnosticValidator>[]
-        {
-            UseMvcDiagnosticValidator.CreateAndInitialize,
-        };
-
-#pragma warning restore RS1008 // Avoid storing per-compilation data into the fields of a diagnostic analyzer.
-
-        public StartupAnalzyer()
+        public override void Initialize(AnalysisContext context)
```

Comment:
The diff is garbo here. I recommend looking at this method in detail because it's elegant and cool..